### DISCUSSION TITLE
Is there a good way to join two+ sheets of the same type and keep properties

### DISCUSSION THREAD

**Discussion by frosencrantz** [saulpw/visidata#1598] [Q&A]
I hit this issue the other day and was trying to read files from several directories so I did this:
```
vd dirpath1 dirpath2 dirpath3
```
I wanted to have all the files in a single sheet.   So I go to SheetsSheet select all the DirSheets and do an append join get a sheet with all the rows of the three dirs.  But I hit return on the rows and instead of opening the file it dives into the row.  Looking at SheetsSheet,  I see the new sheet is a ConcatSheet.

https://asciinema.org/a/XyVmOq3fUcxVOLAEk44qbjcuU

Wondering if when doing an append join for sheets of all of the same type  VisiData could use the same sheet type as the base sheets? 

In this case there are a some workarounds:
```
find dirpath1 dirpath2 dirpath3 | vd -n -f fdir
```

Or another way would be before joining the sheets I can unhide all the columns `gv` so that the  `abspath`  column is showing and use `zo` to open a file of interest.

maybe there is a better way.

**Comment by anjakefala** [comment]
I do not have my full cognitive capabilities right now, so this might be a curveball. But, I believe that Saul created the **Extend** Join specifically for a join type that preserves the properties of one of the sheets. It does not work smoothly for all of the possible combinations, but I think it was made with the **DirSheet** in mind. (I believe it is an outer join that preserves sheet properties.)

Unfortunately, this seems to be one of those Super Features that is under-documented.

**Comment by frosencrantz** [comment]
@anjakefala **Extend** command didn't help.  Since it adds columns from the other sheets. It does not append rows:

 - **Extend**: https://asciinema.org/a/nDbFqqjSSblujYguGCCOJcrcz
    - The result is still a DirSheet.
 - **Outer**: https://asciinema.org/a/y9jNgKdY12KeY0oX5Zp1uaniS
    - The result is a JoinSheet.

So extend does preserve the sheet type, but it doesn't allow adding rows.  Extend does seem useful if there is additional information about the existing rows, but not in this.  Yes, it would be helpful to know these Super Features.  I wasn't aware of that differences of these join types.

**Comment by saulpw** [comment]
This is a good use case, @frosencrantz, and it was easy enough to implement a special case for `append` join of sheets of the same type, as a simple concatenation of rows, instead of a ConcatSheet.
